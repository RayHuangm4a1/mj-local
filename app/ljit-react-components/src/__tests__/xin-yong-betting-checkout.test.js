import React from 'react';
import { shallow, mount, } from 'enzyme';
import XinYongBettingCheckout, { PREFIX_CLASS, } from '../components/xin-yong-betting-checkout';
import ChipGroup from '../components/xin-yong-betting-checkout/chip-group';

describe('XinYongBettingCheckout', () => {
	it('should PREFIX_CLASS to equal ljit-xin-yong-betting-checkout', () => {
		expect(PREFIX_CLASS).toEqual('ljit-xin-yong-betting-checkout');
	});

	it('should handle default props', () => {
		const {
			betCount,
			betAmount,
			balance,
			isSquare,
			inputValue,
			onSubmit,
			onReset,
			onChangeInputValue,
		} = XinYongBettingCheckout.defaultProps;

		expect(betCount).toEqual(0);
		expect(betAmount).toEqual(0);
		expect(balance).toEqual(0);
		expect(isSquare).toEqual(false);
		expect(inputValue).toEqual(0);
		expect(onSubmit).toBeInstanceOf(Function);
		expect(onReset).toBeInstanceOf(Function);
		expect(onChangeInputValue).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const props = {
			betCount: 0,
			betAmount: 0,
			balance: 0,
			isSquare: false,
			onSubmit: () => {},
			onReset: () => {},
			inputValue: 0,
			changeInputValue: () => {},
		};

		const wrapper = shallow(<XinYongBettingCheckout {...props}/>);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-xin-yong-betting-checkout',() => {
		const wrapper = shallow(<XinYongBettingCheckout />);

		expect(wrapper.hasClass('ljit-xin-yong-betting-checkout')).toEqual(true);
	});
	it('should ljit-xin-yong-betting-checkout--square be selectable by isSquare',() => {
		const wrapper = shallow(<XinYongBettingCheckout isSquare={true} />);

		expect(wrapper.hasClass('ljit-xin-yong-betting-checkout--square')).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const props = {
			betCount: 0,
			betAmount: 0,
			balance: 0,
			isSquare: false,
			onSubmit: () => {},
			onReset: () => {},
			inputValue: 0,
			changeInputValue: () => {},
		};
		const wrapper = mount(<XinYongBettingCheckout {...props}/>);

		expect(wrapper.props().betCount).toEqual(props.betCount);
		expect(wrapper.props().betAmount).toEqual(props.betAmount);
		expect(wrapper.props().balance).toEqual(props.balance);
		expect(wrapper.props().isSquare).toEqual(props.isSquare);
		expect(wrapper.props().inputValue).toEqual(props.inputValue);
		expect(wrapper.props().onSubmit).toEqual(props.onSubmit);
		expect(wrapper.props().onReset).toEqual(props.onReset);
		expect(wrapper.props().changeInputValue).toEqual(props.changeInputValue);
	});

	it('should handle onReset event', () => {
		const onReset = jest.fn();
		const wrapper = mount(<XinYongBettingCheckout onReset={onReset} />);
		const resetButton = wrapper.find('.ljit-xin-yong-betting-checkout__reset__button').at(0);

		resetButton.simulate('click');
		expect(onReset).toHaveBeenCalled();
	});

	it('should handle sumbit event', () => {
		const onSubmit = jest.fn();
		const wrapper = mount(<XinYongBettingCheckout onSubmit={onSubmit} />);
		const submitButton = wrapper.find('.ljit-xin-yong-betting-checkout__submit__button').at(0);

		submitButton.simulate('click');
		expect(onSubmit).toHaveBeenCalled();
	});
});

describe('ChipGroup' , () => {
	it('should handle default props', () => {
		const {
			className,
			onClick,
		} = ChipGroup.defaultProps;

		expect(className).toEqual('');
		expect(onClick).toBeInstanceOf(Function);
	});
	it('should render correctly', () => {
		const props = {
			className: 'mock-name',
			onClick: () => {},
		};
		const wrapper = shallow(<ChipGroup {...props}/>);

		expect(wrapper).toMatchSnapshot();
	});
	it('should be selectable by class ljit-xin-yong-betting-checkout__chip',() => {
		const wrapper = shallow(<ChipGroup />);

		expect(wrapper.hasClass('ljit-xin-yong-betting-checkout__chip')).toEqual(true);
	});
	it('should mount in a full DOM', () => {
		const props = {
			className: 'mock-name',
			onclick: () => {},
		};
		const wrapper = mount(<XinYongBettingCheckout {...props}/>);

		expect(wrapper.props().className).toEqual(props.className);
		expect(wrapper.props().onclick).toEqual(props.onclick);
	});
});

