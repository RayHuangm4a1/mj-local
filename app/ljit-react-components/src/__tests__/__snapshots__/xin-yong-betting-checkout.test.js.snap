// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipGroup should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChipGroup
    className="mock-name"
    onClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <chipButton
          onClick={[Function]}
          type="1"
        />,
        <chipButton
          onClick={[Function]}
          type="2"
        />,
        <chipButton
          onClick={[Function]}
          type="5"
        />,
        <chipButton
          onClick={[Function]}
          type="10"
        />,
        <chipButton
          onClick={[Function]}
          type="20"
        />,
        <chipButton
          onClick={[Function]}
          type="50"
        />,
      ],
      "className": "ljit-xin-yong-betting-checkout__chip mock-name",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "ljit-chip-button-1",
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "type": "1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ljit-chip-button-2",
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "type": "2",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ljit-chip-button-5",
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "type": "5",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ljit-chip-button-10",
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "type": "10",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ljit-chip-button-20",
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "type": "20",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ljit-chip-button-50",
        "nodeType": "function",
        "props": Object {
          "onClick": [Function],
          "type": "50",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <chipButton
            onClick={[Function]}
            type="1"
          />,
          <chipButton
            onClick={[Function]}
            type="2"
          />,
          <chipButton
            onClick={[Function]}
            type="5"
          />,
          <chipButton
            onClick={[Function]}
            type="10"
          />,
          <chipButton
            onClick={[Function]}
            type="20"
          />,
          <chipButton
            onClick={[Function]}
            type="50"
          />,
        ],
        "className": "ljit-xin-yong-betting-checkout__chip mock-name",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "ljit-chip-button-1",
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "type": "1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ljit-chip-button-2",
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "type": "2",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ljit-chip-button-5",
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "type": "5",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ljit-chip-button-10",
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "type": "10",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ljit-chip-button-20",
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "type": "20",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ljit-chip-button-50",
          "nodeType": "function",
          "props": Object {
            "onClick": [Function],
            "type": "50",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`XinYongBettingCheckout should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <XinYongBettingCheckout
    balance={0}
    betAmount={0}
    betCount={0}
    changeInputValue={[Function]}
    inputValue={0}
    isSquare={false}
    onChangeInputValue={[Function]}
    onReset={[Function]}
    onSubmit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ljit-xin-yong-betting-checkout__input"
        >
          快选金额
          <InputNumber
            disabled={false}
            formatType=""
            min={0.001}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            readOnly={false}
            step={0.001}
            value={0}
          />
        </div>,
        <ChipGroup
          className=""
          onClick={[Function]}
        />,
        <div
          className="ljit-xin-yong-betting-checkout__balance"
        >
          <div>
            共 
            0
             注，共 
            <DecimalNumber
              className=""
              data={0}
              decimalPlaces={4}
              hasSeparator={true}
              isCurrency={false}
              isPercent={false}
              isTolerance={false}
            />
             元
          </div>
          <div>
            余额 
            <DecimalNumber
              className=""
              data={0}
              decimalPlaces={4}
              hasSeparator={true}
              isCurrency={false}
              isPercent={false}
              isTolerance={false}
            />
          </div>
        </div>,
        <div
          className="ljit-xin-yong-betting-checkout__button"
        >
          <Button
            className="ljit-xin-yong-betting-checkout__reset__button"
            color="warmOrange700"
            disabled={false}
            onClick={[Function]}
            outline="hollow"
          >
            清空选号
          </Button>
          <Button
            className="ljit-xin-yong-betting-checkout__submit__button"
            color="warmOrange700"
            disabled={false}
            onClick={[Function]}
            outline="solid"
          >
            下注
          </Button>
        </div>,
      ],
      "className": "ljit-xin-yong-betting-checkout",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "快选金额",
            <InputNumber
              disabled={false}
              formatType=""
              min={0.001}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder=""
              readOnly={false}
              step={0.001}
              value={0}
            />,
          ],
          "className": "ljit-xin-yong-betting-checkout__input",
        },
        "ref": null,
        "rendered": Array [
          "快选金额",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "formatType": "",
              "min": 0.001,
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "placeholder": "",
              "readOnly": false,
              "step": 0.001,
              "value": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              共 
              0
               注，共 
              <DecimalNumber
                className=""
                data={0}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
               元
            </div>,
            <div>
              余额 
              <DecimalNumber
                className=""
                data={0}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
            </div>,
          ],
          "className": "ljit-xin-yong-betting-checkout__balance",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "共 ",
                0,
                " 注，共 ",
                <DecimalNumber
                  className=""
                  data={0}
                  decimalPlaces={4}
                  hasSeparator={true}
                  isCurrency={false}
                  isPercent={false}
                  isTolerance={false}
                />,
                " 元",
              ],
            },
            "ref": null,
            "rendered": Array [
              "共 ",
              0,
              " 注，共 ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "",
                  "data": 0,
                  "decimalPlaces": 4,
                  "hasSeparator": true,
                  "isCurrency": false,
                  "isPercent": false,
                  "isTolerance": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " 元",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "余额 ",
                <DecimalNumber
                  className=""
                  data={0}
                  decimalPlaces={4}
                  hasSeparator={true}
                  isCurrency={false}
                  isPercent={false}
                  isTolerance={false}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "余额 ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "",
                  "data": 0,
                  "decimalPlaces": 4,
                  "hasSeparator": true,
                  "isCurrency": false,
                  "isPercent": false,
                  "isTolerance": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              className="ljit-xin-yong-betting-checkout__reset__button"
              color="warmOrange700"
              disabled={false}
              onClick={[Function]}
              outline="hollow"
            >
              清空选号
            </Button>,
            <Button
              className="ljit-xin-yong-betting-checkout__submit__button"
              color="warmOrange700"
              disabled={false}
              onClick={[Function]}
              outline="solid"
            >
              下注
            </Button>,
          ],
          "className": "ljit-xin-yong-betting-checkout__button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "清空选号",
              "className": "ljit-xin-yong-betting-checkout__reset__button",
              "color": "warmOrange700",
              "disabled": false,
              "onClick": [Function],
              "outline": "hollow",
            },
            "ref": null,
            "rendered": "清空选号",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "下注",
              "className": "ljit-xin-yong-betting-checkout__submit__button",
              "color": "warmOrange700",
              "disabled": false,
              "onClick": [Function],
              "outline": "solid",
            },
            "ref": null,
            "rendered": "下注",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ljit-xin-yong-betting-checkout__input"
          >
            快选金额
            <InputNumber
              disabled={false}
              formatType=""
              min={0.001}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder=""
              readOnly={false}
              step={0.001}
              value={0}
            />
          </div>,
          <ChipGroup
            className=""
            onClick={[Function]}
          />,
          <div
            className="ljit-xin-yong-betting-checkout__balance"
          >
            <div>
              共 
              0
               注，共 
              <DecimalNumber
                className=""
                data={0}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
               元
            </div>
            <div>
              余额 
              <DecimalNumber
                className=""
                data={0}
                decimalPlaces={4}
                hasSeparator={true}
                isCurrency={false}
                isPercent={false}
                isTolerance={false}
              />
            </div>
          </div>,
          <div
            className="ljit-xin-yong-betting-checkout__button"
          >
            <Button
              className="ljit-xin-yong-betting-checkout__reset__button"
              color="warmOrange700"
              disabled={false}
              onClick={[Function]}
              outline="hollow"
            >
              清空选号
            </Button>
            <Button
              className="ljit-xin-yong-betting-checkout__submit__button"
              color="warmOrange700"
              disabled={false}
              onClick={[Function]}
              outline="solid"
            >
              下注
            </Button>
          </div>,
        ],
        "className": "ljit-xin-yong-betting-checkout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "快选金额",
              <InputNumber
                disabled={false}
                formatType=""
                min={0.001}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder=""
                readOnly={false}
                step={0.001}
                value={0}
              />,
            ],
            "className": "ljit-xin-yong-betting-checkout__input",
          },
          "ref": null,
          "rendered": Array [
            "快选金额",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "formatType": "",
                "min": 0.001,
                "onBlur": [Function],
                "onChange": [Function],
                "onFocus": [Function],
                "placeholder": "",
                "readOnly": false,
                "step": 0.001,
                "value": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                共 
                0
                 注，共 
                <DecimalNumber
                  className=""
                  data={0}
                  decimalPlaces={4}
                  hasSeparator={true}
                  isCurrency={false}
                  isPercent={false}
                  isTolerance={false}
                />
                 元
              </div>,
              <div>
                余额 
                <DecimalNumber
                  className=""
                  data={0}
                  decimalPlaces={4}
                  hasSeparator={true}
                  isCurrency={false}
                  isPercent={false}
                  isTolerance={false}
                />
              </div>,
            ],
            "className": "ljit-xin-yong-betting-checkout__balance",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "共 ",
                  0,
                  " 注，共 ",
                  <DecimalNumber
                    className=""
                    data={0}
                    decimalPlaces={4}
                    hasSeparator={true}
                    isCurrency={false}
                    isPercent={false}
                    isTolerance={false}
                  />,
                  " 元",
                ],
              },
              "ref": null,
              "rendered": Array [
                "共 ",
                0,
                " 注，共 ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "",
                    "data": 0,
                    "decimalPlaces": 4,
                    "hasSeparator": true,
                    "isCurrency": false,
                    "isPercent": false,
                    "isTolerance": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " 元",
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "余额 ",
                  <DecimalNumber
                    className=""
                    data={0}
                    decimalPlaces={4}
                    hasSeparator={true}
                    isCurrency={false}
                    isPercent={false}
                    isTolerance={false}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "余额 ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "",
                    "data": 0,
                    "decimalPlaces": 4,
                    "hasSeparator": true,
                    "isCurrency": false,
                    "isPercent": false,
                    "isTolerance": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                className="ljit-xin-yong-betting-checkout__reset__button"
                color="warmOrange700"
                disabled={false}
                onClick={[Function]}
                outline="hollow"
              >
                清空选号
              </Button>,
              <Button
                className="ljit-xin-yong-betting-checkout__submit__button"
                color="warmOrange700"
                disabled={false}
                onClick={[Function]}
                outline="solid"
              >
                下注
              </Button>,
            ],
            "className": "ljit-xin-yong-betting-checkout__button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "清空选号",
                "className": "ljit-xin-yong-betting-checkout__reset__button",
                "color": "warmOrange700",
                "disabled": false,
                "onClick": [Function],
                "outline": "hollow",
              },
              "ref": null,
              "rendered": "清空选号",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "下注",
                "className": "ljit-xin-yong-betting-checkout__submit__button",
                "color": "warmOrange700",
                "disabled": false,
                "onClick": [Function],
                "outline": "solid",
              },
              "ref": null,
              "rendered": "下注",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
